browser/net/brave_ad_block_tp_network_delegate_helper_unittest.cc

```diff
@@ -43,3 +53,79 @@ TEST(BraveAdBlockTPNetworkDelegateHelperTest, DevToolURL) {
   EXPECT_TRUE(request_info->new_url_spec.empty());
   EXPECT_EQ(rc, net::OK);
 }
+
+class TestingBraveComponentUpdaterDelegate : public BraveComponent::Delegate {
+ public:
+  TestingBraveComponentUpdaterDelegate()
+      : task_runner_(base::ThreadPool::CreateSequencedTaskRunner({})) {}
+  ~TestingBraveComponentUpdaterDelegate() override {}
+
+  using ComponentObserver = update_client::UpdateClient::Observer;
+  // brave_component_updater::BraveComponent::Delegate implementation
+  void Register(const std::string& component_name,
+                const std::string& component_base64_public_key,
+                base::OnceClosure registered_callback,
+                BraveComponent::ReadyCallback ready_callback) override {}
+  bool Unregister(const std::string& component_id) override { return true; }
+  void OnDemandUpdate(const std::string& component_id) override {}
+
+  void AddObserver(ComponentObserver* observer) override {}
+  void RemoveObserver(ComponentObserver* observer) override {}
+
+  scoped_refptr<base::SequencedTaskRunner> GetTaskRunner() override {
+    return task_runner_;
+  }
+
+  const std::string locale() const override { return "en"; }
+  PrefService* local_state() override { return pref_service_.get(); }
+
+ private:
+  scoped_refptr<base::SequencedTaskRunner> task_runner_;
+
+  std::unique_ptr<PrefService> pref_service_;
+
+  DISALLOW_COPY_AND_ASSIGN(TestingBraveComponentUpdaterDelegate);
+};
+
+// Serves as `next_callback` for the request handler
+void OnSuccess() {}
+
+TEST(BraveAdBlockTPNetworkDelegateHelperTest, SimpleBlocking) {
+  content::BrowserTaskEnvironment task_environment;
+
+  // It appears that g_browser_process automatically gets created for unit
+  // tests. It should be possible to make that work for g_brave_browser_process
+  // as well.
+  TestingBraveBrowserProcess::CreateInstance();
+
+  auto brave_component_updater_delegate =
+      std::make_unique<TestingBraveComponentUpdaterDelegate>();
```

Comment:
I made `AdBlockService::Start()` work without any browser dependencies by implementing a custom component updater delegate class just for testing. That appears to be the "correct" way of doing it :smile: